client/src/search/Typeahead/TypeaheadMenu.js

```diff
@@ -0,0 +1,110 @@
+import { ListGroup, ListGroupItem } from "react-bootstrap";
+
+import { create_text_maker_component } from "../../components";
+
+import text from "./Typeahead.yaml";
+
+const { text_maker, TM } = create_text_maker_component(text);
+
+export class TypeaheadMenu extends React.Component {
+  handleWindowClick = (e) => {
+    const { hide_menu } = this.props;
+    if (!document.getElementsByClassName("rbt")[0].contains(e.target)) {
+      hide_menu();
+    }
+  };
+  componentDidMount() {
+    window.addEventListener("click", this.handleWindowClick);
+  }
+
+  render() {
+    const {
+      search_text,
+      queried_results,
+      config_groups,
+      on_select_item,
+      hide_menu,
+    } = this.props;
+
+    const total_matching_results = queried_results.length;
+
+    let index_key_counter = 0;
+
+    if (_.isEmpty(queried_results)) {
+      return (
+        <ListGroup className="rbt-menu dropdown-menu show">
+          <ListGroupItem className="dropdown-header">
+            {text_maker("no_matches_found")}
+          </ListGroupItem>
+        </ListGroup>
+      );
+    } else {
+      return (
+        <ListGroup className="rbt-menu dropdown-menu show">
+          {_.chain(queried_results)
+            .groupBy("config_group_index")
+            .thru((grouped_results) => {
+              return [
+                <ListGroupItem
+                  key={`header-pagination-info`}
+                  className="dropdown-header"
+                >
+                  <TM
+                    k="paginate_status"
+                    args={{
+                      total_matching_results,
+                    }}
+                  />
+                </ListGroupItem>,
+                ..._.flatMap(grouped_results, (results, group_index) => [
+                  <ListGroupItem
+                    key={`header-${group_index}`}
+                    className="dropdown-header"
+                  >
+                    {config_groups[group_index].group_header}
+                  </ListGroupItem>,
+                  <div
+                    key={`group-${group_index}`}
+                    role="group"
+                    aria-label={config_groups[group_index].group_header}
```

Comment:
Screen reader tries to count the groups, but with the current layout that count is wrong. Navigation skips the first and last "group" (the result count and close button respectively), so the first group is announced as "2 of N" and the last is "N-1 of N". The group names are also not read out until after the full name of the first option, which is likely both easy to miss and kind of confusing.

The first and last items should probably not be groups. Pretty sure if the groups themselves are focusable then it will announce the name of the group and give the option to enter in to it or continue to the next group. That'd be better both for announcing group names and navigation, I think.

Reply:
Seems like all children of `ul` are to be counted as their own groups...

I'll have to look into workarounds for this.